Hosting Overview: Adjust domains table styles

xavier-lc opened the pull request:
Related to https://github.com/Automattic/dotcom-forge/issues/6384

## Proposed Changes

* This PR addresses some design inconsistencies noticed by @mmtr on https://github.com/Automattic/wp-calypso/pull/89584#issuecomment-2063547956

## Testing Instructions

* Open the live preview
* Go to `/hosting-overview`
* Check that the domains table matches the desktop (PZtnZ1nHhnBrsuXDoMJHPD-fi-446%3A153354) and mobile (IQgFHqGiN0eXkPn7Qkth4s-fi-6%3A8860) designs

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

Commit:
`DomainsTable`: Add `className` prop

Commit:
Rotate toggle button

Commit:
Style primary domain label

Commit:
`DomainsTable`: Add `useMobileCards` prop

Commit:
Use a wider mobile breakpoint on the `ActiveDomainsCard`

Commit:
Only show domain info on mobile

Commit:
Remove padding

Commit:
Adjust spacing

Commit:
Hide bulk actions

Commit:
Adjust font

Commit:
Add border to table top

Commit:
Adjust padding

davemart-in commented:
Mobile design looks solid.

Here's what I'm seeing for the desktop design:

![CleanShot 2024-04-25 at 15 24 08@2x](https://github.com/Automattic/wp-calypso/assets/5634774/d93fda2e-003f-4711-9ce3-574b4c074af2)

A couple of things:

- Not sure why there are check boxes
- Looks like longer domain names are compressing each of the other columns. Thoughts on putting a max width on the domain column and then using a ... for domains that are longer? We could add a title attribute with the full domain that they could see on mouseover.

![CleanShot 2024-04-25 at 15 27 01@2x](https://github.com/Automattic/wp-calypso/assets/5634774/832d0d56-04fe-4a9d-ace1-5aa9689d2714)

Commit:
`DomainsTable` Add `hideCheckbox` prop so it's easier to get rid of it via css

Commit:
Hide checkboxes

Commit:
Make sure the proper css rules have precedence

xavier-lc commented:
> * The font-size of the domains looks a tad larger than what I'm seeing in the Figma file (14px or `$font-body-small`)

I found that sometimes the stylesheet order changes, fixed on https://github.com/Automattic/wp-calypso/pull/89806/commits/19f82f2e72ed5b2964c6d7ac64661e5f42b7caa0

> * Not sure why there are check boxes

Oh, it seems to be related to subdoomains, but I don't have any so I wasn't seeing them. I've addressed it on https://github.com/Automattic/wp-calypso/pull/89806/commits/083481d223442f910920eb388e2cfaf0a3c5a194, can you check if it looks good now?

> * Looks like longer domain names are compressing each of the other columns.

Do you think it's still necessary now that the font size is smaller?

davemart-in reviewed (approved):
Thanks for addressing those. I think we're good now with the font being smaller. This looks ready to go!
